### PR TITLE
Storybook in core

### PR DESCRIPTION
**Pull request by ilariaventurini**
This PR adds Storybook in the core package.

Each demo object has this structure:

    {
      options: ...,
      data: ...,
      chartType: ...,
      isDemoExample: boolean,
    }

`isDemoExample` allows you to choose whether the example should be visible only in the storybook or also in the demo page.

So, if:
- `isDemoExample = true` -> the example is inserted both in the demo page and in the Storybook
- `isDemoExample = false` -> the example is inserted only in the Storybook.

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/IBM/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/core/.storybook/config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { configure } from "@storybook/html";
+import { setOptions } from "@storybook/addon-options";
+
+setOptions({
+  name: "Carbon Charts - HTML Wrappers",
```
Comment (theiliad) [inline packages/core/.storybook/config.js]:
```suggestion
  name: "Carbon Charts - Vanilla",
```

File: packages/core/.storybook/main.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const path = require("path");
+
+module.exports = {
+  addons: [
+    '@storybook/preset-typescript', 
+  ]
+}
```
Comment (theiliad) [inline packages/core/.storybook/main.js]:
```suggestion
}

```

File: packages/core/.storybook/main.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const path = require("path");
+
+module.exports = {
+  addons: [
+    '@storybook/preset-typescript',
```
Comment (theiliad) [inline packages/core/.storybook/main.js]:
```suggestion
    '@storybook/preset-typescript'
```

File: packages/core/src/styles/styles.scss [hunk 4]
```diff
@@ -18,4 +19,4 @@ import one of the styles-[theme name here].scss files";
 
 svg.#{$prefix}--#{$charts-prefix}--chart-svg {
 	overflow: visible;
-}
+}
```
Comment (theiliad) [inline packages/core/src/styles/styles.scss]:
```suggestion
}

```

**Comment by theiliad** [comment]
Good to review

**Comment by natashadecoste** [comment]
the paths seems to be broken for me within the demo folder - it doesn't resolve the paths for any of these  `import { Tools } from "@carbon/charts/tools";` ... if i change it back to the full path it all runs nicely 🧐

**Comment by theiliad** [comment]
> the paths seems to be broken for me within the demo folder - it doesn't resolve the paths for any of these `import { Tools } from "@carbon/charts/tools";` ... if i change it back to the full path it all runs nicely 🧐

@natashadecoste you'd need to `yarn build` first in `core`

**Comment by natashadecoste** [comment]
> > the paths seems to be broken for me within the demo folder - it doesn't resolve the paths for any of these `import { Tools } from "@carbon/charts/tools";` ... if i change it back to the full path it all runs nicely 🧐
> 
> @natashadecoste you'd need to `yarn build` first in `core`

@theiliad ah i see, i guess i just did a `demo:build`

**Pull request merged**